Framework/Algorithms/src/XDataConverter.cpp

```diff
@@ -103,7 +103,7 @@ void XDataConverter::setXData(API::MatrixWorkspace_sptr outputWS,
         }
       }
     }
-    outputWS->setX(index, m_cachedX);
+    outputWS->setSharedX(index, m_cachedX);
   } else {
     const MantidVec &xBoundaries = inputWS->readX(index);
     MantidVec &xPoints = outputWS->dataX(index);
```

Comment:
There are a few uses of the old `readX` and `dataX` that need to be replaced, here and above.